internal/commands/user/list/command_test.go

```diff
@@ -0,0 +1,192 @@
+package list
+
+import (
+	//"errors"
+
+	//"strings"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestUserListSetup(t *testing.T) {
+	t.Run("Should construct a Realm client with the configured base url", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		cmd := &command{inputs: inputs{}}
+		assert.Nil(t, cmd.realmClient)
+
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+	})
+}
+
+func TestUserListHandler(t *testing.T) {
+	testApp := realm.App{
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+
+	newMockClient := func() mock.RealmClient {
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			testApp.GroupID = filter.GroupID
+			testApp.ID = filter.App
```

Comment:
[question] why are we overwriting the `testApp` details here? if for testing purposes, i'd rather us have a new variable(s) like `capturedAppFilter`/`actualAppFilter` or something similar

Reply:
I was doing that for testing purposes.  Not so much to capture the values there, but to return an app with the correct ID and GroupID.  I added capturing variables when I find the users with the app.  LMK if you don't want this

Reply:
we shouldn't need to be mutating a `testApp` struct's data to ensure a test passes (especially a `testApp` struct that's outside of this test closure and is used in downstream/subsequent tests)

in general, we want to use mock functions to either return static data/errors, or simply pass-through/capture the data it receives through args.  in this case, I think just adding `ID` and `GroupID` fields to the `testApp` initialization up front would be a fine alternative.

also, can we add a test (or build on to this one) to make sure we are passing in the appropriate `realm.UserFilter` based on the inputs provided?